src/Riok.Mapperly/Symbols/NonEmptyMemberPath.cs

```diff
@@ -0,0 +1,23 @@
+using Microsoft.CodeAnalysis;
+
+namespace Riok.Mapperly.Symbols;
+
+public class NonEmptyMemberPath : MemberPath
+{
+    public NonEmptyMemberPath(IReadOnlyList<IMappableMember> path)
```

Comment:
I understand what you are doing here... But doesn't this introduce weird semantics: an instance of `MemberPath` can be empty but doesn't have to (the class also contains a lot of logic which only is needed if it contains a non-empty path)? 🤔 

On a second thought: Wouldn't it be easier to special case the empty member path instead of the non-empty one? Idea to discuss:
* Remove the `NonEmptyMemberPath` and revert usage back to `MemberPath`
* Add a new `RootObjectMemberPath : MemberPath` (or maybe name it  `EmptyMemberPath`)
* In `SymbolAccessor.TryFindMemberPath` if the path is empty return a new instance of the `RootObjectMemberPath` instead of the usual `MemberPath`. Pass the containing `type` to it.
* Set the `MemberPath.MemberType` property to non-nullable and virtual
* In `RootObjectMemberPath` overwrite the `MemberType` to return the type of the root which was passed in via constructor.
* `MemberPath` would simply throw if `Member` or `MemberType` is accessed on an empty `MemberPath` unless it is a `RootObjectMemberPath` (which would be expected).

WDYT?

Reply:
I like your updated approach 😊 The only point that looks kinda weird to me is the new naming of the `GetterMemberPath` and `SetterMemberPath`. For example `*Mapping` classes storing and exposing `*Builder` properties looks strange to me (e.g. `NullMemberMapping`).
Overall I think I'd prefer keeping the old names `GetterMemberPath` and `SetterMemberPath`. This should also drastically reduce the size of this PR.

Reply:
`MemberPathGetter/Setter` indicates that the `MemberPathSetter` would provide functionality to `set` a `MemberPath` which isn't the case either. For me the best fit (although not perfect) is still `GetterMemberPath` and `SetterMemberPath`.